### PR TITLE
Update Resize (opset 11) layer to support scales option when dims are…

### PR DESCRIPTION
**Pull request by qant-um**
Resize layer (version 11) will generate automatically multiple layers using Shape, Slice, Concat whereas it supports also the scales and sizes options to realize this operation. In the case where input shape is undefined (-1), this method is interesting to be applied but otherwise, inference will be faster with only one node of Resize. Backend unit test are already efficient to cover this usecase.
reference: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Resize-11

### CODE REVIEW

File: tf2onnx/onnx_opset/nn.py [hunk 1]
```diff
@@ -1391,20 +1391,37 @@ def version_11(cls, ctx, node, **kwargs):
...
-        size_int64 = ctx.make_node("Cast", [node.input[1]], attr={"to": onnx_pb.TensorProto.INT64})
-        concat_shape = ctx.make_node("Concat", [sliced_shape.output[0], size_int64.output[0]], {'axis': 0})
-        resize_inputs = [
-            input_nchw.output[0],
-            roi.output[0],
-            const_empty_float.output[0],
-            concat_shape.output[0]
-        ]
+        shape = ctx.get_shape(node.input[0])
+        if shape and shape[2] != -1 and shape[1] != -1 and node.inputs[1].is_const():
```
Comment (fatcat-z) [inline tf2onnx/onnx_opset/nn.py:1396]: The shape comes from the input of Resize. Is it possible the rank is less than 3 so shape[2] doesn't exist?

Comment (qant-um) [reply tf2onnx/onnx_opset/nn.py:1396]: Hi,
This line is a copy of a condition of the version 10 ... but you are right, if this appends there will be an issue.
From my first understanding, shape is (B, C, H, W) so you mean the graph could be in another shape dimension. So, a condition should check that shape is on 4 dims before entering in this instruction block like this for example:
`if shape and len(shape) == 4 and and shape[2] != -1 and shape[1] != -1 and node.inputs[1].is_const()`
or "playing" with the array indexes:
`if shape and shape[-2] != -1 and shape[-1] != -1 and node.inputs[1].is_const()`

Thanks for your review again,

Comment (qant-um) [reply tf2onnx/onnx_opset/nn.py:1396]: Do you mean, this is ok for you ? I do not know exactly how the process works. Do you attend an action from myself ?
or the merge will be done by you once tests are passed ?

**Review by fatcat-z — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**